static/js/settings_bots.js

```diff
@@ -39,8 +42,21 @@ function render_bots() {
             is_active: elem.is_active,
             zuliprc: 'zuliprc', // Most browsers do not allow filename starting with `.`
         });
+        is_all_inactive_bots = is_all_inactive_bots && !elem.is_active;
     });
 
+    if (page_params.realm_bot_creation_policy !==
+        exports.bot_creation_policy_values.admins_only.code) {
+        if (all_bots_for_current_user.length === 0 || is_all_inactive_bots) {
+            $("#add-a-new-bot-form").show();
+            $("#active_bots_list").hide();
+            $("#inactive_bots_list").hide();
+            $("#bots_lists_navbar .active").removeClass("active");
+            $("#bots_lists_navbar .add-a-new-bot-tab").addClass("active");
+            return;
```

Comment:
This code feels really redundant and error-prone; maybe extract the block repeated with the next section to a function?